react/src/utils/tableHelpers.ts

```diff
@@ -108,3 +111,91 @@ export const sortQueryResult = (queryResult: VariantQueryDataResult[]) => {
     );
     return sortedQueryResult;
 };
+
+export const isHeterozygous = (zygosity: String | null | undefined) => {
+    if (zygosity?.toLowerCase().includes('het')) {
+        return true;
+    }
+    return false;
+};
+
+export const isHomozygous = (zygosity: String | null | undefined) => {
+    if (zygosity?.toLowerCase().includes('hom')) {
+        return true;
+    }
+    return false;
+};
+
+// 1, Sort queryResult in ascending order according to variant's ref, alt, start, end.
+// 2, Flatten data and compute values as needed (note that column display formatting function should not alter values for ease of export). Assign uniqueId, homozygousCount, heterozygousCount to each row.
+export const prepareData = (
+    queryResult: VariantQueryDataResult[]
+): [ResultTableColumns[], number[]] => {
+    const sortedQueryResult = sortQueryResult(queryResult);
+
+    const result: Array<ResultTableColumns> = [];
+
+    // contains indices of first encountered rows that represent unique variants.
+    const uniqueVariantIndices: Array<number> = [];
+
+    var currVariant = {} as Variant;
+    var currUniqueId = 0;
+    var currRowId = 0;
+    var currHomozygousCount = 0;
+    var currHeterozygousCount = 0;
+
+    sortedQueryResult.forEach(d => {
+        const { ref, alt, start, end } = d.variant;
+
+        if (JSON.stringify(currVariant) !== JSON.stringify({ ref, alt, start, end })) {
```

Comment:
Yes, because it is being done in a loop. You have the overhead of the conversion rather than comparing four properties in memory. Furthermore (unrelated to performance), this relies on initializing `currVariant` below in the same order as the construction above.

Reply:
Ok got it. I've changed it to direct comparisons between fields.